src/Jackett.Common/Utils/Extensions.cs

```diff
@@ -26,7 +26,7 @@ public static class XElementExtension
 
     public static class TaskExtensions
     {
-        public static Task<IEnumerable<TResult>> Until<TResult>(this IEnumerable<Task<TResult>> tasks, TimeSpan timeout)
+        public static Task<IEnumerable<TResult>> Until<TResult>(this IReadOnlyCollection<Task<TResult>> tasks, TimeSpan timeout)
```

Comment:
Here we had quite dangerous multiple enumeration. We enumerate tasks the first time by calling `Task.WhenAll(tasks)`, and the second time to select results. It worked fine as we convert all our tasks to lists. But potentially we were possible to do something like this:
```
var tasks = entities.Select(entity => Save(entity));
var results = tasks.Until(TimeSpan.FromSeconds(3)).ToArray();
```

In this scenario we will call `entity => Save(entity)` twice for each entity

Reply:
Another option - leave IEnumerable, but convert to List / Array here

Reply:
@mynameisbogdan, Only semantically. The caller sees that we have no intention of modifying the collection.
We still have the ability to pass arrays and lists, but we won't be able to call Add or Remove.

Let's change to ICollection if you don't like it

Reply:
@mynameisbogdan, Okay, I wrote a small example that shows why we should avoid IEnumerable here:
```
async Task<int> SaveAsync(int num)
{
    Console.WriteLine("SaveAsync");
    await Task.Delay(1000);
    return num;
}

var tasks = Enumerable.Range(0, 3).Select(SaveAsync);
var results = await tasks.Until(TimeSpan.FromHours(1));

Console.WriteLine($"result.Count: {results.Count()}");
```

What do you think this program will output? It seems that the output will consist of three "SaveAsync" and the result will consist of three numbers. But no, here's what happens:
```
SaveAsync
SaveAsync
SaveAsync
SaveAsync
SaveAsync
SaveAsync
result.Count: 0
```

But if we change it to IReadOnlyCollection (or any other collection):
```
SaveAsync
SaveAsync
SaveAsync
result.Count: 3
```